### PR TITLE
feat: telemetry lib [WPB-11977]

### PR DESCRIPTION
**Pull request by olafsulich**
## Description

Telemetry library, Countly wrapper.

I've only added methods that are used in current scenarios within the Web App and TM. To be compatible with the current implementation, I've mapped many methods 1:1. In the future we can adapt it to our needs, simplify it, change it, or add new methods from Countly.

Usage (drafts):
- [refactor: use telemetry lib (wire-team-settings)](https://github.com/wireapp/wire-team-settings/pull/8548/files#diff-76010a4a5b5ff94e165a675fe54b5867f52afb95661568bc68c09cdc9bf5a445)
- [refactor: use telemetry lib (wire-webapp)](https://github.com/wireapp/wire-team-settings/pull/8548/files#diff-76010a4a5b5ff94e165a675fe54b5867f52afb95661568bc68c09cdc9bf5a445)

ADR: [001-initialize-library](https://github.com/wireapp/wire-webapp/pull/18285/files#diff-f5d532f11887b4763660a02cea44e276ff2dfe3d867e8098e0e08efac5d76f2d)

## Checklist

- [x] mentions the JIRA issue in the PR name (Ex. [WPB-423])
- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [ ] If it is a core feature, unit tests have been added;

[WPB-423]: https://wearezeta.atlassian.net/browse/WPB-423?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/telemetry/src/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  /**
+   * The an object with key/value pairs to report with the event as segments
+   */
+  segmentation?: Record<string, SegmentationType>;
+};
+
+/**
+ * Configuration options for initializing the analytics library.
+ */
+interface InitilaizeConfig {
```
Comment (aweiss-dev) [inline packages/telemetry/src/telemetry.ts:57]: Typo

File: packages/telemetry/src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+};
+
+/**
+ * Removes consent features from the analytics library.
+ *
+ * Use this to inform provider of the features the user has withdrawn consent for.
+ *
+ * @param {TelemetryConsentFeatures[]} features - An array of consent features to remove.
+ */
+export const removeConsentFeatures = (features: TelemetryConsentFeatures[]): void => {
```
Comment (aweiss-dev) [inline packages/telemetry/src/telemetry.ts:276]: I would add two additional methods,
addAllConsentFeatures()
removeAllConsentFeatures()

**Review by aweiss-dev — APPROVED** [review]

**Review by e-maad — APPROVED** [review]
Amazing work 👏 🚀

File: packages/telemetry/src/telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+     *
+     * In offline mode, data is collected but not sent to the server until offline mode is disabled.
+     * If consent was enabled, it must be reestablished after enabling offline mode.
+     *
+     * @default true
+     */
+    offline_mode?: boolean;
+  };
+}
+
```
[the remark this answers is not in the document]
Comment (olafsulich) [reply packages/telemetry/src/telemetry.ts:153]: Yeah, probably a good idea to split this file a little. But I'm not a big fan of separating type from implementation. 

I'd rather split it by functionality, e.g. create a file for consents features, offline mode, etc.

But for now, IMO it's not necessary. Let's see how we're gonna grow this library in the future, and when it becomes a problem (readbility, cognitive load, etc.) then split it up somehow.

WDYT?

[the remark this answers is not in the document]
Comment (olafsulich) [reply packages/telemetry/src/telemetry.ts:153]: @przemvs sounds good?

**Pull request merged**